### PR TITLE
Refactored helper func error handling in tests

### PR DESCRIPTION
**Pull request by jmguzik**
**What this PR does / why we need it**:
Remove expects from helper funcs in test

**Which issue(s) this PR fixes** 
Related to (partly fixes) #4101

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by jmguzik** [comment]
@phoracek Could you mark here ok-to-test? It would ease up further work.

**Comment by jmguzik** [comment]
@EdDev as you reviewed previous commit related to this issue I would invite you to review this one

**Comment by jmguzik** [comment]
@phoracek @EdDev PTAL

File: tests/vmi_multus_test.go
Comment (EdDev) [inline tests/vmi_multus_test.go:1388]: I suggest you keep the original loop here and only return the error.
It can be implemented with `wait.PollImmediate` (you should find usage even in this test file).

File: tests/vmi_multus_test.go
Comment (EdDev) [inline tests/vmi_multus_test.go:1436]: Is the returned error contain enough information to understand where it failed?

File: tests/vmi_multus_test.go
Comment (EdDev) [inline tests/vmi_multus_test.go:1449]: Same, is the info enough to understand where we failed?

**Review by EdDev — CHANGES_REQUESTED** [review]
Thank you for this nice PR.

Could you please use in this PR only the first commit, which seems network tests related and the rest post on a separate PR? They are also simpler.
I have just went over the 1st commit in the review.

File: tests/vmi_multus_test.go
Comment (jmguzik) [reply tests/vmi_multus_test.go:623]: Previously there was a function used only once that wrapped this. Since function was used only once, I just removed it and in place pasted content

File: tests/vmi_multus_test.go
Comment (jmguzik) [reply tests/vmi_multus_test.go:1436]: I think I checked it and would say yes, but will revisit

File: tests/vmi_multus_test.go
Comment (jmguzik) [reply tests/vmi_multus_test.go:1449]: same here

File: tests/vmi_multus_test.go
Comment (jmguzik) [reply tests/vmi_multus_test.go:1388]: So get rid of eventually and use wait.PollImmediate?

**Comment by jmguzik** [comment]
> Could you please use in this PR only the first commit, which seems network tests related, and the rest post on a separate PR? They are also simpler.

What are your rules then? This commit is under the same issue #4101. That is why they are grouped together. Every commit is handling errors in the different test files. You would like to see an isolated test case file in every PR?

I understand that if PR tackles the issue, commits can be grouped. My mistake?

File: tests/vmi_multus_test.go
Comment (EdDev) [reply tests/vmi_multus_test.go:623]: I think the confusing thing is that you do this in two tests, and in the rest you use this one.
So I guess it makes sense, but you need to remove the ones that appear in the tests as well.

I would also place it in a `BeforeEach` instead, to make sure no one mutates this data.  In general we now try to avoid using `BeforeAll` because it has no counter-part (`AfterAll`), resulting in flaky results.

File: tests/vmi_multus_test.go
Comment (EdDev) [reply tests/vmi_multus_test.go:1388]: Yea, unless you see a problem here.

**Review by EdDev — COMMENTED** [review]
> @EdDev What are your rules then? PR and commits are under the same issue #4101. That is why they are grouped together. Every commit is handling errors in the different test files. You would like to see an isolated test case file in every PR?
> 
> I understand that if PR tackles the issue, commits can be grouped. My mistake?

The project is split by sigs and not all are agreeing on the same patterns. I prefer you split because then we do not need to negotiate it too much for the first commit, as the network sig is in favor of this approach (extracting the asserts).
So there are no rules here, just an attempt to make sure we are not blocked by objections from other reviewers/approvers.

It should be easy to split it using git, but if you prefer to keep it intact I am fine with it. We can split if someone objects.

**Comment by jmguzik** [comment]
> It should be easy to split it using git, but if you prefer to keep it intact I am fine with it. We can split if someone objects.

Yep, git is not the problem. The problem is the general stability of the CI. As you may saw, this PR was affected over the days by multiple flake tests, needs of rebasing and merge, etc, and yet I would say that changes are pretty simple (non-functional). Further splitting may lead to a 2x these kind of problems for fairly simple changes. For me it would be the only reason to little object here, but I might just simply follow your suggestion.

File: tests/vmi_multus_test.go
Comment (jmguzik) [reply tests/vmi_multus_test.go:1388]: I will try but I had a problem with this one. There was no eventually accepting two args and a map was needed. I had several different thoughts that looked strange. Moving return to a function that and later invoking ready lambda seemed a good solution at a time. I will try your approach but ultimately I can't promise to not land with that one in the end.

File: tests/vmi_multus_test.go
Comment (jmguzik) [reply tests/vmi_multus_test.go:623]: So, my understanding is that you want this to be moved to BeforeEach, right?

Comment (EdDev) [reply tests/vmi_multus_test.go:623]: That one was one, yes. The other was not to call it also in the other two tests.. Hope I did not missed some hidden reason there.

**Comment by EdDev** [comment]
> Yep, git is not the problem. The problem is the general stability of the CI. As you may saw, this PR was affected over the days by multiple flake tests, needs of rebasing and merge, etc, and yet I would say that changes are pretty simple (non-functional). Further splitting may lead to a 2x these kind of problems for fairly simple changes. For me it would be the only reason to little object here, but I might just simply follow your suggestion.

Yes, our CI is very flaky :/
It is a matter of balancing, so I am fine with you continuing like this. If someone will object to something in the other commits, we can split then (or manage to convince them otherwise).

File: tests/vmi_multus_test.go
Comment (EdDev) [inline tests/vmi_multus_test.go:1390]: The error here is: "no virt-handler pods detected".

Note that with errors, it is common not to start with capital letters, as these many times chained to other error text.

File: tests/vmi_multus_test.go
Comment (EdDev) [reply tests/vmi_multus_test.go:1484]: Have you tried the `wait.PollImmediate` approach?
I am re-reading this again and it really complicates things and is not intuitive.

Expecting `configureNodeNetwork` to return a function that returns a daemonset is too odd.
It does not read right to me.

How about this...
```
nodes := tests.GetAllSchedulableNodes(virtClient)
err := wait.PollImmediate(time.Second, time.Minute, func() (bool, error) {
    c, e := getNetworkConfigDaemonSet()
    if e != nil {
        return false, e
    }
    return int(c) == len(nodes.Items), nil
})
if err != nil {
    return err
}
```

Comment (jmguzik) [reply tests/vmi_multus_test.go:1484]: Yes, I tried, and it did not work. I will evaluate your proposal as my code differed a little bit from yours. If this would fail, I just skip converting this function.
It's your choice in the end, but I tend to disagree that it looks too odd. I see similar constructs there and there all the way across the projects. In the end, is just a matter of convention.

File: tests/vmi_multus_test.go
Comment (jmguzik) [reply tests/vmi_multus_test.go:1390]: Yes, overlooked that. Thanks

**Pull request closed without merging**
